BeerOverflowWindowsApp/BarProviders/GetBarListFacebook.cs

```diff
@@ -21,52 +21,67 @@ class GetBarListFacebook : IBeerable
-            {
-                throw exception;
-            }
-            return result;
+            var link = string.Format(_apiLink, _accessToken, latitude, longitude, radius, _requestedFields, _category);
+            var jsonStream = GetJsonStream(link);
+            var barList = JsonConvert.DeserializeObject<PlacesResponse>(jsonStream).data;
+            return barList;
+        }
+
+        public async Task<List<BarData>> GetBarsAroundAsync(string latitude, string longitude, string radius)
+        {
+            RegexTools.LocationDataTextIsCorrect(latitude, longitude, radius);
+            var placeList = await GetBarDataAsync(latitude, longitude, radius);
+            var barList = PlaceListToBarList(placeList);
+            return barList;
         }
 
-        private List<BarData> FacebookDataToBars(PlacesResponse resultData)
+        private async Task<List<Place>> GetBarDataAsync(string latitude, string longitude, string radius)
+        {
+            RegexTools.LocationDataTextIsCorrect(latitude, longitude, radius);
+            var link = string.Format(_apiLink, _accessToken, latitude, longitude, radius, _requestedFields, _category);
+            var jsonStream = await GetJsonStreamAsync(link);
+            var barList = JsonConvert.DeserializeObject<PlacesResponse>(jsonStream).data;
+            return barList;
+        }
+
+        private List<BarData> PlaceListToBarList(IEnumerable<Place> resultData)
         {
             var barList = new List<BarData>();
             
-            foreach (var result in resultData.data)
+            foreach (var result in resultData)
             {
                 if (result.restaurant_specialties != null && result.restaurant_specialties.drinks != 1) continue;
                 var barCategories = result.category_list.Select(category => category.name).ToList();
 
                 if (!HasCategories(barCategories, _allowedCategories)) continue;
                 if (HasCategories(barCategories, _bannedCategories)) continue;
-                var newBar = new BarData()
-                {
-                    Title = result.name,
-                    Latitude = result.location.latitude,
-                    Longitude = result.location.longitude,
-                    Ratings = new List<int>()
-                };
+                var newBar = PlaceToBar(result);
                 barList.Add(newBar);
             }
             return barList;
         }
 
-        private bool HasCategories(IEnumerable<string> categories, string bannedCategoriesInString)
+        private static BarData PlaceToBar(Place place)
+        {
+            var newBar = new BarData
+            {
+                Title = place.name,
+                BarId = place.name,   // Temporary solution until we decide on BarId 
+                Latitude = place.location.latitude,
+                Longitude = place.location.longitude,
+                Ratings = new List<int>()
+            };
+            return newBar;
```

Comment:
You could make it: 
return new BarData{ ...};
Not sure if its more efficient, but  I think i looks a bit nicer.